narwhals/dataframe.py

```diff
@@ -288,7 +294,7 @@ def gather_every(self, n: int, offset: int = 0) -> Self:
             self._compliant_frame.gather_every(n=n, offset=offset)
         )
 
-    def join_asof(
+    def join_asof(  # noqa: C901
```

Comment:
@dangotbanned this new check pushes the method over the complexity limit. Rather extract this out to its own fn or leave the `# noqa: C901`?